Data/ByteString/Lazy.hs

```diff
@@ -563,29 +578,25 @@ compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - fromIntegral (S.le
    n <= length t = compareLength t n /= LT
   #-}
 
+mapAccumLChunks :: (acc -> S.ByteString -> (acc, S.ByteString)) -> acc -> ByteString -> (acc, ByteString)
+mapAccumLChunks function accumulator = fmap (L.foldr Chunk Empty) . L.mapAccumL function accumulator . foldrChunks (:) [ ]
+
+mapAccumRChunks :: (acc -> S.ByteString -> (acc, S.ByteString)) -> acc -> ByteString -> (acc, ByteString)
+mapAccumRChunks function accumulator = fmap (L.foldl (flip Chunk) Empty) . L.mapAccumL function accumulator . foldlChunks (flip (:)) [ ]
```

Comment:
Could you possibly use `L.mapAccumR` and avoid `flip`?

Reply:
Possibly. I am not sure it would have the right strictness properties. This is all murky waters yet. I need to add more strictness checks.

Reply:
Let's do it and see if there are any complaints.